.github/workflows/ci-test.yml

```diff
@@ -0,0 +1,152 @@
+          EOF
+          chmod 755 external/bin/gcc external/bin/g++
+          echo '${{ github.workspace }}/external/bin' >> $GITHUB_PATH
+
+      - name: 'Install dependencies (macOS)'
+        if: runner.os == 'macOS' && matrix.branch == 'devel'
+        run: |
+          brew install openssl@1.1
+          ln -s $(brew --prefix)/opt/openssl/lib/libcrypto.1.1.dylib /usr/local/lib
+          ln -s $(brew --prefix)/opt/openssl/lib/libssl.1.1.dylib /usr/local/lib/
+
+      - name: MSYS2 (Windows i386)
+        if: runner.os == 'Windows' && matrix.target.cpu == 'i386'
+        uses: msys2/setup-msys2@v2
+        with:
+          path-type: inherit
+          msystem: MINGW32
+          install: >-
+            base-devel
+            git
+            mingw-w64-i686-toolchain
+
+      - name: MSYS2 (Windows amd64)
+        if: runner.os == 'Windows' && matrix.target.cpu == 'amd64'
+        uses: msys2/setup-msys2@v2
+        with:
+          path-type: inherit
+          install: >-
+            base-devel
+            git
+            mingw-w64-x86_64-toolchain
+
+      - name: Restore Nim DLLs dependencies (Windows) from cache
+        if: runner.os == 'Windows'
+        id: windows-dlls-cache
+        uses: actions/cache@v4
+        with:
+          path: external/dlls-${{ matrix.target.cpu }}
+          key: 'dlls-${{ matrix.target.cpu }}'
+
+      - name: Install DLLs dependencies (Windows)
+        if: >
+          steps.windows-dlls-cache.outputs.cache-hit != 'true' &&
+          runner.os == 'Windows'
+        run: |
+          mkdir -p external
+          curl -L "https://nim-lang.org/download/windeps.zip" -o external/windeps.zip
+          7z x -y external/windeps.zip -oexternal/dlls-${{ matrix.target.cpu }}
+
+      - name: Path to cached dependencies (Windows)
+        if: >
+          runner.os == 'Windows'
+        run: |
+          echo "${{ github.workspace }}/external/dlls-${{ matrix.target.cpu }}" >> $GITHUB_PATH
+
+      - name: Derive environment variables
+        run: |
+          if [[ '${{ matrix.target.cpu }}' == 'amd64' ]]; then
+            PLATFORM=x64
+          else
+            PLATFORM=x86
+          fi
+          echo "PLATFORM=$PLATFORM" >> $GITHUB_ENV
+
+          ncpu=
+          MAKE_CMD="make"
+          case '${{ runner.os }}' in
+          'Linux')
+            ncpu=$(nproc)
+            ;;
+          'macOS')
+            ncpu=$(sysctl -n hw.ncpu)
+            ;;
+          'Windows')
+            ncpu=$NUMBER_OF_PROCESSORS
+            MAKE_CMD="mingw32-make"
+            ;;
+          esac
+          [[ -z "$ncpu" || $ncpu -le 0 ]] && ncpu=1
+          echo "ncpu=$ncpu" >> $GITHUB_ENV
+          echo "MAKE_CMD=${MAKE_CMD}" >> $GITHUB_ENV
+
+      - name: Download Nimble and install Nim
+        run: |
+          curl -O -L -s -S https://github.com/nim-lang/nimble/releases/download/latest/nimble-linux_x64.tar.gz
+          tar xvfz nimble-linux_x64.tar.gz
+          ./nimble install nim
```

Comment:
broadly, we should be able to use `nimble setup` here, which _should_ install a package-local version of nim that `nimble test` later uses

Reply:
Nimble setup worked out https://github.com/status-im/nim-stew/actions/runs/9663859361/job/26657108122

Reply:
nimble install nim@1.6.20 also worked out when full version tag was specified.
https://github.com/status-im/nim-stew/actions/runs/9667227572/job/26668548913

Reply:
nimble install nim@2.1.1 ended well, but similar situation with older Nim. This time instead of Nim 2.1.1 older Nim 2.0.6 was effectively used to run the tests.
https://github.com/status-im/nim-stew/actions/runs/9667421954/job/26669209430

Reply:
What would we like to see here ? I would personally see the easiest to have Nimble which could install older Nim, and fix versions of older Nim which could follow reasonably young Nimble through back porting. 

Example:
Nim 1.6 is the last major.minor for v1 and its fix versions 1.6.20 and so on would allow being managed by Nimble which could manage Nim v2 versions.

This would relieve devs who need to stick to Nim 1.6 for very long time. Maybe could add 2+ years of lifetime?

If this is too complex or slow on innovation, we might follow the rustc and cargo approach. Moving faster through versions and having same versioning for compiler and package manager. Rust has also editions to create "promise" across many versions. Not sure if any promise is kept for Nim major versions ?

Reply:
I've found a discussion on the related topic in "very old high priority" feature request 654 at issues at nimble at minlang. I feel like we would need to start collecting numbers and data points to give sea of opinions clarity they deserve. :)

Reply:
I've compiled report about what I am doing. To tidy up and have something presentable for community. https://www.notion.so/Nim-Tooling-Test-Reports-7b25687860ae41afaa2d4f22af8d07f1?pvs=4 

What should we do next ? @arnetheduck

Reply:
Add Tests05 and Tests06 to compare download and installation/setup time to the report.